cmd/setProperties.go

```diff
@@ -0,0 +1,153 @@
+// Copyright © 2017 Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// TODO check the problem with TAGS not working with SAS (mismatched permissions)
+package cmd
+
+import (
+	"fmt"
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+func (raw *rawCopyCmdArgs) setMandatoryDefaultsForSetProperties() {
+	raw.blobType = common.EBlobType.Detect().String()
+	raw.md5ValidationOption = common.DefaultHashValidationOption.String()
+	raw.s2sInvalidMetadataHandleOption = common.DefaultInvalidMetadataHandleOption.String()
+	raw.forceWrite = common.EOverwriteOption.True().String()
+	raw.preserveOwner = common.PreserveOwnerDefault
+}
+
+func (cca *CookedCopyCmdArgs) checkIfChangesPossible() error {
+	// tier can't be set on files
+
+	if cca.FromTo.From() == common.ELocation.File() && (cca.blockBlobTier != common.EBlockBlobTier.None() || cca.pageBlobTier != common.EPageBlobTier.None()) {
+		return fmt.Errorf("changing tier is not available for File Storage")
+	}
+	return nil
+}
+
+func (cca *CookedCopyCmdArgs) makeEnumAndPerformChecks() error {
+	if cca.blockBlobTier != common.EBlockBlobTier.None() || cca.pageBlobTier != common.EPageBlobTier.None() {
+		if cca.FromTo.From() == common.ELocation.File() {
+			return fmt.Errorf("tier cannot be set upon files")
+		}
+		cca.propertiesToTransfer |= common.ESetPropertiesFlags.SetTier()
+	}
+	if cca.metadata != "" {
+		cca.propertiesToTransfer |= common.ESetPropertiesFlags.SetMetadata()
+	}
+	if cca.blobTags != nil {
+		// the fact that fromto is not filenone is taken care of by the cook function
+		cca.propertiesToTransfer |= common.ESetPropertiesFlags.SetBlobTags()
+	}
+	return cca.checkIfChangesPossible()
```

Comment:
Redundant code. This check is being performed at line 49 as well

Reply:
removed